### ISSUE TITLE
Testing controllers with 3 different headsets : not working

### ISSUE THREAD

**Issue by AkinaChinami** [KitwareMedical/SlicerVirtualReality#116]
We have tested different versions of Slicer with 3 different headsets: Oculus Quest 2, Oculus Rift and Valve Index. These tests have been made to help developers find out where the problem might be coming from. Here's our test report.

For **Oculus Quest 2 & Oculus Rift**:

1. _Version 4.11_:

- **Fly** : move around in space.
   _Right joystick touch_ : move forward enable to fly forward / move backward enable to fly backward
- **Transform entire scene** : scale / translate / rotate.
   Keeping _Trigger touch_ or _Grip button_ depressed on both controllers and: 
   - Move closer together or move farther apart enable to scale size of the entire scene.
   - Translate controllers in parallel up or down or left or right or forward or backward enable to translate the entire scene in these directions.
   - Pivot controllers around enable to rotate the entire scene

- **Transform objects : translate / rotate a selected object**
   Pressing _Grip Button_ or _Trigger Touch_ when a controller is inside a selectable object enable to grab the object then :
   - Move controllers closer together or farther apart enable to scale size of the object
   - translate controllers in parallel up or down or left or right or forward or backward enable to translate the object

2. _Version 5.2.2:_

   _Error message_ : "the action manifest for slicerapp-real was not found"
   json files are copied in the local folder of slicer and in the extension folder

- **Fly :** 
   _Right joystick touch_ move forward enable to fly forward / move backward enable to fly backward
- **Transform entire scene is disabled** : _trigger touch_ depressed on both controllers, and nothing happened (disable to translate, disable to scale, disable to rotate)
- **Transform objects is disabled** : _Grip button_ depressed make the volume disappeared

3. _Version 5.3.0-2023-05-31:_

   _Error message_ : "the action manifest for slicerapp-real was not found"
   json files are copied in the local folder of slicer and in the extension folder
   json files are still not found by slicer-app real

_________________________________________________________________________________________________________________________________________________

For **Valve Index:**

1. _Version 4.11_:

- **Fly** : move around in space
   _Right joystick touch_ or the _Trackpad touch_: move forward enable to fly forward / move backward enable to fly backward

- **Transform entire scene** : scale / translate / rotate
   Keeping _Trigger touch_ or _A button_ depressed on both controllers and :
   - move closer together or move farther apart enable to scale size of the entire scene
   - translate controllers in parallel up or down or left or right or forward or backward enable to translate the entire scene in these directions
   - pivot controllers around enable to rotate the entire scene

- **Transform objects** : translate / rotate a selected object
   Pressing _Trigger Touch_ or _A button_ when a controller is inside a selectable object enable to grab the object then :
   - move controllers closer together or farther apart enable to scale size of the object
   - translate controllers in parallel up or down or left or right or forward or backward enable to translate the object

2. _Version 5.2.2:_

   _Error message_ : "the action manifest for slicerapp-reak was not found"
   json files are copied in the local folder of slicer and in the extension folder

- **Fly** : 
   _Right joystick touch_ move forward enable to fly forward / move backward enable to fly backward
   The _trackpad touch_ does not any effect

- **Transform entire scene is disabled** : _trigger touch_ depressed on both controllers, and nothing happened (disable to translate, disable to scale, disable to rotate)

- **Transform objects is disabled** : _A button_ depressed on both controllers make the volume disappeared. If only depressed on one controller, nothing happened

3. _Version 5.3.0-2023-05-31:_

   _Error message_ : "the action manifest for slicerapp-real was not found"
   json files are copied in the local folder of slicer and in the extension folder
   json files are still not found by slicer-app real

**Comment by LucasGandel** [comment]
Thanks @AkinaChinami for the outstanding report!
From my perspective, `Version 5.3.0-2023-05-31` should fix most of the interaction issues reported with `Version 5.2.2.`.
However, I confirm `Version 5.3.0-2023-05-31` is impacted by the action manifest issue, see #115

**Comment by cpinter** [comment]
Is there anything from this list that still doesn't work in Slicer 5.3.0 2023-06-12? (Except the manifest renaming hack, which has its own ticket #115)

**Comment by AkinaChinami** [comment]
@cpinter After several tests to be sure, it seems that all interactions work, thanks for your help!

**Comment by jcfr** [comment]
Cross-referencing - This issue is related to:
* https://github.com/KitwareMedical/SlicerVirtualReality/issues/91

**Comment by cpinter** [comment]
Thanks for testing @AkinaChinami ! I'm closing this ticket then. Please reopen this if you don't agree, or open a new ticket if you find specific issues.
